cmd/kops-controller/controllers/legacy_node_controller.go

```diff
@@ -199,27 +198,28 @@ func (r *LegacyNodeReconciler) getInstanceGroupForNode(ctx context.Context, node
 	return r.loadNamedInstanceGroup(instanceGroupName)
 }
 
-// loadCluster loads a kops.Cluster object from a vfs.Path
-func (r *LegacyNodeReconciler) loadCluster(p vfs.Path) (*kops.Cluster, error) {
+// loadCluster loads a api.Cluster object from a vfs.Path
+func (r *LegacyNodeReconciler) loadCluster(p vfs.Path) (*api.Cluster, error) {
 	ttl := time.Hour
 
 	b, err := r.cache.Read(p, ttl)
 	if err != nil {
 		return nil, fmt.Errorf("error loading Cluster %q: %v", p, err)
 	}
 
-	o, _, err := kopscodecs.Decode(b, nil)
+	defaultGVK := api.SchemeGroupVersion.WithKind("Cluster")
+	o, _, err := kopscodecs.Decode(b, &defaultGVK)
```

Comment:
Why do we need to pass a `defaultGVK`?

Reply:
The intent is that this triggers the version-conversion logic, so we get either v1alpha2 or v1alpha3, but I'll verify this is doing what I think it's doing.

Reply:
By my reading, `kopscodecs.Decode()` always decodes to the unversioned protocol because it uses `Codecs.UniversalDecoder(kops.SchemeGroupVersion)`.

The version in `b` always takes precedence over `defaultGVK`. The completed-cluster is always written versioned, so I'm not seeing how `defaultGVK` is ever used.

Reply:
I'll give it a go, to be more homogenous with the other code.  We're type checking it anyway, so if we start seeing the type check fail with the wrong package, we'll know we need a deeper look!